native/cocos/bindings/event/EventDispatcher.cpp

```diff
@@ -237,6 +244,58 @@ void EventDispatcher::dispatchKeyboardEvent(const KeyboardEvent &keyboardEvent)
     EventDispatcher::doDispatchJsEvent(eventName, args);
 }
 
+void EventDispatcher::dispatchControllerEvent(const ControllerEvent &controllerEvent) {
+    se::AutoHandleScope scope;
+    if (!jsControllerEventArray) {
+        jsControllerEventArray = se::Object::createArrayObject(0);
+        jsControllerEventArray->root();
+    }
+
+    const char *eventName = "onControllerInput";
+    if (controllerEvent.type == ControllerEvent::Type::HANDLE) {
+        eventName = "onHandleInput";
+    }
+    uint32_t    controllerIndex = 0;
+    jsControllerEventArray->setProperty("length", se::Value(static_cast<uint32_t>(controllerEvent.controllerInfos.size())));
+
+    for (const auto &controller : controllerEvent.controllerInfos) {
+        auto *jsController = se::Object::createPlainObject();
```

Comment:
Use `se::HandleObject jsController{se::Object::createPlainObject()};` for plain JS object instead to avoid memory leaks.